simnode/generator.py

```diff
@@ -85,50 +87,166 @@ def _calculate_parallel_coords(offset: float, line_width: float) \
+                    for p in cutting_points:
+                        # cython int32 to int
+                        p = int(p)
+                        if p > 0:
+                            try:
+                                # calculate parallel offset for each road piece
+                                road_lnstr = LineString(original_line.coords[previous_p: p]).parallel_offset(offset[i])
+                                if offset_sub_lines.__len__() == 0:
+                                    offset_sub_lines = road_lnstr.coords.xy
+                                    # needs to be reversed for positive offset (why?)
+                                    if offset[i] > 0:
+                                        offset_sub_lines[0].reverse()
+                                        offset_sub_lines[1].reverse()
+                                else:
+                                    # try except not pretty, but problems with empty line string segments solved
+                                    try:
+                                        line_str_0 = (road_lnstr.coords.xy)[0]
+                                        line_str_1 = (road_lnstr.coords.xy)[1]
+                                        # needs to be reversed for positive offset (why?)
+                                        if offset[i] > 0:
+                                            line_str_0.reverse()
+                                            line_str_1.reverse()
+                                        offset_sub_lines[0].extend(line_str_0)
+                                        offset_sub_lines[1].extend(line_str_1)
+                                    except Exception:
+                                        # this break fixes failing calculation, if a line segment is empty, but
+                                        # it is really ugly
+                                        break
+                            except (NotImplementedError, Exception):  # FIXME Where is TopologyException
+                                _logger.exception("Creating an offset line for lane markings failed")
+                                return None
+                        previous_p = p
+                        i += 1
+                    if offset_sub_lines.__len__() > 1:
+                        # softens stair stepping if the road width changes fast
+                        # TODO tolerance parameter may need to be adjusted
+                        # could be expensive --> replace with #coords = offset_sub_lines
+                        point_list = list(map(lambda i: (offset_sub_lines[0][i], offset_sub_lines[1][i]),
+                                              range(0, offset_sub_lines[0].__len__()-1)))
+                        lstr = LineString(point_list)
+                        lstr = lstr.simplify(tolerance=0.13)
+                        coords = lstr.coords.xy
+
+                        # for plotting the generated road, useful for debugging
```

Comment:
Please do not include debug code into productive code eventhough it might be helpful for debugging puproses.